src/Mapping.elm

```diff
@@ -0,0 +1,463 @@
+
+
+formatPriceItems : List PriceItem -> List String
+formatPriceItems items =
+    List.map formatPriceItem items
+
+
+
+-- Exercise 11: Multiply by Index
+-- Task: Multiply each number in a list by its index.
+
+
+multiplyByIndex : List Int -> List Int
+multiplyByIndex list =
+    List.indexedMap multiplyAtIndex list
+
+
+multiplyAtIndex : Int -> Int -> Int
+multiplyAtIndex index value =
+    index * value
+
+
+
+-- Exercise 12: Generate HTML Tags
+-- Task: Wrap each string in a list with HTML <p> tags.
+
+
+wrapWithPTags : String -> String
+wrapWithPTags str =
+    "<p>" ++ str ++ "</p>"
+
+
+wrapListWithPTags : List String -> List String
+wrapListWithPTags strings =
+    List.map wrapWithPTags strings
+
+
+
+-- Exercise 13: Calculate Logarithms
+-- Task: Calculate the natural logarithm of each number in a list.
+
+
+calculateLog : Float -> Float
+calculateLog number =
+    logBase e number
+
+
+calculateLogs : List Float -> List Float
+calculateLogs numbers =
+    List.map calculateLog numbers
+
+
+
+-- Exercise 14: Replace Empty with Placeholder
+-- Task: Replace empty strings in a list with "Placeholder".
+
+
+replaceEmpty : String -> String
+replaceEmpty str =
+    if String.isEmpty str then
+        "Placeholder"
+
+    else
+        str
+
+
+replaceEmptyList : List String -> List String
+replaceEmptyList list =
+    List.map replaceEmpty list
+
+
+
+-- Exercise 15: Normalize Text
+-- Task: Sanitize strings in a list by trimming
+-- whitespace and converting text to lowercase.
+
+
+normalizeText : String -> String
+normalizeText text =
+    text
+        |> trim
+        |> toLower
+
+
+normalizeTexts : List String -> List String
+normalizeTexts texts =
+    List.map normalizeText
+        texts
+
+
+
+-- Exercise 16: Find Lengths
+-- Task: Return the length of each string in a list.
+
+
+lengthsOfStrings : List String -> List Int
+lengthsOfStrings strings =
+    List.map String.length strings
+
+
+
+-- Exercise 17: Append "day"
+-- Task: Append "day" to the end of each string in a list
+-- representing shortened or non-standard days of the week.
+
+
+appendDay : String -> String
+appendDay day =
+    day ++ "day"
+
+
+appendDays : List String -> List String
+appendDays days =
+    List.map appendDay days
+
+
+
+-- Exercise 18: Convert Extended String to Boolean
+-- Task: Convert each string in a list to its boolean
+-- representation, considering extended variations such as
+-- "true", "yes", "y", "1" for True and "false", "no", "n", "0"
+-- for False. Case should not affect the conversion.
+
+
+stringToBool : String -> Bool
+stringToBool str =
+    case toLower str of
+        "true" ->
+            True
+
+        "yes" ->
+            True
+
+        "y" ->
+            True
+
+        "1" ->
+            True
+
+        "false" ->
+            False
+
+        "no" ->
+            False
+
+        "n" ->
+            False
+
+        "0" ->
+            False
+
+        _ ->
+            False
+
```

Comment:
This needs some work.

The question says that case should not matter.

So, "True", "Yes", and "Y" should evaluate to `True`.

Right now, they'd evaluate to `False`.

Reply:
Also, you can collapse all of the `False` branches into one.

Since you've already specified everything that should be `True`, every other thing should be `False`. You can use `_` to represent every other thing!